src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -831,12 +871,26 @@ public int hashCode() {
     }
 
     private void logNodeMessage(FlowNode node) {
+        List<String> nesting = getNesting(node);
+        if (!nesting.isEmpty() && nesting.get(nesting.size() - 1).equals(node.getId())) {
+            // For a BlockStartNode, we do not want to hide itself.
+            nesting = new ArrayList<>(nesting.subList(0, nesting.size() - 1));
+        }
+        try {
+            listener.annotate(new NestingNote(nesting));
+        } catch (IOException x) {
+            LOGGER.log(Level.WARNING, null, x);
+        }
         WorkflowConsoleLogger wfLogger = new WorkflowConsoleLogger(listener);
         String prefix = getLogPrefix(node);
+        String text = node.getDisplayFunctionName();
+        if (node instanceof BlockStartNode) {
```

Comment:
Would be useful to be able to collapse `AtomNode` output as well: for example, long `sh` steps, or `build` if [JENKINS-26124](https://issues.jenkins-ci.org/browse/JENKINS-26124) is implemented.